Description: Allow to build without pulling git-version and concat_const
Author: Mike Hommey <mh@glandium.org>
Origin: upstream
Applied-Upstream: 1a916c16d02aebf36091ef906897e7a4c87530b0
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: git-cinnabar/Cargo.toml
===================================================================
--- git-cinnabar.orig/Cargo.toml
+++ git-cinnabar/Cargo.toml
@@ -86,6 +86,7 @@ features = [
 
 [dependencies.concat_const]
 version = "0.1"
+optional = true
 
 [dependencies.cstr]
 version = "0.2.10"
@@ -119,6 +120,7 @@ version = "0.1"
 
 [dependencies.git-version]
 version = "0.3"
+optional = true
 
 [dependencies.hex]
 version = "0.4"
@@ -203,10 +205,12 @@ version = "2.0"
 [features]
 compile_commands = []
 curl-compat = ["rustflags"]
-default = ["version-check"]
+default = ["full-version", "version-check"]
+full-version = ["dep:concat_const", "dep:git-version"]
 gitdev = []
 self-update = [
     "shared_child",
+    "dep:concat_const",
     "dep:tar",
     "dep:xz2",
 ]
Index: git-cinnabar/src/version.rs
===================================================================
--- git-cinnabar.orig/src/version.rs
+++ git-cinnabar/src/version.rs
@@ -171,18 +171,25 @@ macro_rules! full_version {
 mod static_ {
     use clap::crate_version;
     // Work around https://github.com/rust-lang/rust-analyzer/issues/8828 with `as cat`.
+    #[cfg(feature = "full-version")]
     use concat_const::concat as cat;
+    #[cfg(feature = "full-version")]
     use git_version::git_version;
 
     #[cfg(any(feature = "version-check", feature = "self-update"))]
     use super::BuildBranch;
 
     pub const SHORT_VERSION: &str = crate_version!();
+    #[cfg(feature = "full-version")]
     const GIT_VERSION: &str = git_version!(
         args = ["--always", "--match=nothing/", "--abbrev=40", "--dirty=m"],
         fallback = "",
     );
+    #[cfg(feature = "full-version")]
     pub const MODIFIED: bool = matches!(GIT_VERSION.as_bytes().last(), Some(b'm'));
+    #[cfg(not(feature = "full-version"))]
+    pub const MODIFIED: bool = false;
+    #[cfg(feature = "full-version")]
     pub const BUILD_COMMIT: &str = unsafe {
         // Subslicing is not supported in const yet.
         std::str::from_utf8_unchecked(std::slice::from_raw_parts(
@@ -190,12 +197,16 @@ mod static_ {
             GIT_VERSION.len() - if MODIFIED { 1 } else { 0 },
         ))
     };
+    #[cfg(not(feature = "full-version"))]
+    pub const BUILD_COMMIT: &str = "";
 
     #[cfg(any(feature = "version-check", feature = "self-update"))]
     pub const BUILD_BRANCH: BuildBranch = BuildBranch::from_version(SHORT_VERSION);
 
-    //    #[allow(clippy::const_is_empty)]
+    #[cfg(feature = "full-version")]
     pub const FULL_VERSION: &str = full_version!(SHORT_VERSION, BUILD_COMMIT, MODIFIED, cat);
+    #[cfg(not(feature = "full-version"))]
+    pub const FULL_VERSION: &str = SHORT_VERSION;
 }
 
 fn value<'a, T: 'a + ConfigType + ?Sized, F: FnOnce(T::Owned) -> Option<T::Owned>>(
